src/xpk/core/blueprint/blueprint_generator.py

```diff
@@ -354,6 +360,289 @@ def _get_a3_mega_blueprint_dependencies(
+            }],
+        },
+    )
+    net_1_id = f"{cluster_name}-a3u-net-1"
+    gpu_net_1 = DeploymentModule(
+        id=net_1_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//modules/network/vpc?ref=e0c690b",
+        settings={
+            "network_name": f"{cluster_name}-a3u-net-1",
+            "mtu": mtu_size,
+            "subnetworks": [{
+                "subnet_name": f"{cluster_name}-a3u-sub-1",
+                "subnet_region": region,
+                "subnet_ip": "192.168.64.0/18",
+            }],
+            "firewall_rules": [{
+                "name": f"{cluster_name}-a3u-internal-1",
+                "ranges": ["192.168.0.0/16"],
+                "allow": [
+                    {"protocol": "tcp", "ports": ["0-65535"]},
+                    {"protocol": "udp", "ports": ["0-65535"]},
+                    {"protocol": "icmp"},
+                ],
+            }],
+        },
+    )
+    rma_net_id = f"{cluster_name}-a3u-rdma-net"
+    rma_net = DeploymentModule(
+        id=rma_net_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//community/modules/network/rdma-vpc?ref=98c49fe",
+        settings={
+            "network_name": f"{cluster_name}-a3u-rdma-net",
+            "mtu": mtu_size,
+            "network_profile": f"https://www.googleapis.com/compute/beta/projects/{project_id}/global/networkProfiles/{zone}-vpc-roce",
+            "network_routing_mode": "REGIONAL",
+            "subnetworks_template": {
+                "name_prefix": f"{cluster_name}-a3u-rdma-sub",
+                "count": 8,
+                "ip_range": "192.168.128.0/18",
+                "region": region,
+            },
+        },
+    )
+    cluster_id = f"{cluster_name}-a3-ultragpu-cluster"
+    a3_ultra_cluster = DeploymentModule(
+        id=cluster_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//modules/scheduler/gke-cluster?ref=e0c690b",
+        use=[net_0_id],
+        settings={
+            "release_channel": "RAPID",
+            "prefix_with_deployment_name": False,
+            "name_suffix": cluster_name,
+            "system_node_pool_machine_type": system_node_pool_machine_type,
+            "system_node_pool_disk_size_gb": system_node_pool_disk_size_gb,
+            "system_node_pool_taints": [],
+            "enable_dcgm_monitoring": True,
+            "enable_gcsfuse_csi": True,
+            "enable_private_endpoint": False,
+            "autoscaling_total_min_nodes": autoscaling_total_min_nodes,
+            "initial_node_count": num_nodes,
+            "spot": spot,
+            "master_authorized_networks": [{
+                "cidr_block": auth_cidr,
+                "display_name": "kubectl-access-network",
+            }],
+            "system_node_pool_node_count": [{
+                "total_min_nodes": system_node_pool_min_node_count,
+                "total_max_nodes": 1000,
+            }],
+            "maintenance_exclusions": [{
```

Comment:
do we need this? if yes why till 2025-12-22?

Reply:
It is good to have it to avoid constant cluster upgrades. This values are copied from ct, I may remove it from our bluperint and users will set it

Reply:
removed